python/ray/data/datasource/sql_datasource.py

```diff
@@ -23,12 +24,38 @@ def _cursor_to_block(cursor) -> Block:
 
 @PublicAPI(stability="alpha")
 class SQLDatasource(Datasource):
+
+    MAX_ROWS_PER_WRITE = 128
+
     def __init__(self, connection_factory: Callable[[], Connection]):
         self.connection_factory = connection_factory
 
     def create_reader(self, sql: str) -> "Reader":
         return _SQLReader(sql, self.connection_factory)
 
+    def write(
+        self,
+        blocks: Iterable[Block],
+        ctx: TaskContext,
+        sql: str,
+    ) -> WriteResult:
+        with _connect(self.connection_factory) as cursor:
+            for block in blocks:
+                block_accessor = BlockAccessor.for_block(block)
+
+                values = []
+                for row in block_accessor.iter_rows(public_row_format=False):
+                    values.append(tuple(row.values()))
+                    assert len(values) <= self.MAX_ROWS_PER_WRITE, len(values)
```

Comment:
I thought @raulchen wants us to do a size-based check, not number-of-rows-based check, right?

Reply:
Discussed offline, we can do the number-of-rows-based check for now.